src/integrations/integr_postgres.rs

```diff
@@ -0,0 +1,117 @@
+use crate::tools::tools_description::Tool;
+use crate::at_commands::at_commands::AtCommandsContext;
+use crate::call_validation::{ChatContent, ChatMessage, ChatUsage};
+use crate::call_validation::ContextEnum;
+use async_trait::async_trait;
+use serde_json::Value;
+use std::collections::HashMap;
+use std::sync::Arc;
+use tokio::sync::Mutex as AMutex;
+use tokio::process::Command;
+use serde_yaml;
+use std::path::PathBuf;
+use tracing::info;
+
+pub struct ToolPostgres {
+    connection_string: String,
+    psql_binary_path: PathBuf,
+}
+
+impl ToolPostgres {
+    pub fn new_if_configured(integrations_value: &serde_yaml::Value) -> Option<Self> {
+        let postgres = integrations_value.get("postgres")?;
+        let connection_string = postgres.get("connection_string")?.as_str()?.to_string();
+        let psql_binary_path = postgres.get("psql_binary_path")?.as_str()?;
+
+        Some(ToolPostgres {
+            connection_string,
+            psql_binary_path: PathBuf::from(psql_binary_path),
+        })
+    }
+
+    async fn run_psql_command(&self, query: &str) -> Result<String, String> {
+        let mut cmd = Command::new(&self.psql_binary_path);
+        cmd.arg(&self.connection_string)
+            .arg("ON_ERROR_STOP=1")
+            .arg("-c")
+            .arg(query);
+
+        let output = cmd.output().await
+            .map_err(|e| format!("Failed to execute psql command: {}", e))?;
+
+        if output.status.success() {
+            Ok(String::from_utf8_lossy(&output.stdout).to_string())
+        } else {
+            Err(format!("psql command failed: {}", String::from_utf8_lossy(&output.stderr)))
+        }
+    }
+}
+
+#[async_trait]
+impl Tool for ToolPostgres {
+    async fn tool_execute(
+        &mut self,
+        _ccx: Arc<AMutex<AtCommandsContext>>,
+        tool_call_id: &String,
+        command: &HashMap<String, Value>,
+    ) -> Result<(bool, Vec<ContextEnum>), String> {
+        let query = match command.get("command") {
+            Some(Value::String(v)) => v.clone(),
+            Some(v) => return Err(format!("argument `command` is not a string: {:?}", v)),
+            None => return Err("Command is empty".to_string()),
+        };
+        
+        let result = self.run_psql_command(&query).await?;
+
+        let mut results = vec![];
+        results.push(ContextEnum::ChatMessage(ChatMessage {
+            role: "tool".to_string(),
+            content: ChatContent::SimpleText(serde_json::to_string(&result).unwrap()),
+            tool_calls: None,
+            tool_call_id: tool_call_id.clone(),
+            ..Default::default()
+        }));
+        Ok((true, results))
```

Comment:
can psql write output in json?

Reply:
Nope 
<img width="606" alt="image" src="https://github.com/user-attachments/assets/27df9783-e845-4d1b-ab68-d167da5c0f16">